postmortem.py

```diff
@@ -0,0 +1,59 @@
+import logging
+
+import pytest
+from pytest import CollectReport, StashKey
+
+_phase_report_key = StashKey[dict[str, CollectReport]]()
+
+
+@pytest.hookimpl(wrapper=True, tryfirst=True)
+def pytest_runtest_makereport(item: pytest.Item, call: pytest.CallInfo[None]):
+    """
+    Make the report for a step available in the item's stash, so that
+    fixtures can read the result in their teardown phase.
+    Heavily inspired by: https://docs.pytest.org/en/latest/example/simple.html#making-test-result-information-available-in-fixtures
+    Implements the runtest_makereport-hook: https://docs.pytest.org/en/latest/reference/reference.html#std-hook-pytest_runtest_makereport
+
+    :param item: The item the report is generated for.
+    :param call: CallInfo for this phase
+    :return: The report re received from the previous hook.
+    """
+    rep = yield
+
+    # store test results for each phase of a call, which can
+    # be "setup", "call", "teardown"
+    item.stash.setdefault(_phase_report_key, {})[rep.when] = rep
+    return rep
+
+
+_pm_logger = logging.getLogger("post-mortem")
+
+
+@pytest.fixture(autouse=True)
+def pm_system(request: pytest.FixtureRequest, shell, record_property):
+    """
+    Gathers post-mortem information on the state of the system in case the
+    test fails during the "call" phase.
+    """
+    yield
+
+    post_mortem_info = {}
+
+    def _log(command: str):
+        stdout = shell.run_check(command)
+        _pm_logger.warning(command)
+        for line in stdout:
+            _pm_logger.warning(line)
+        post_mortem_info[command] = stdout
+
+    report = request.node.stash[_phase_report_key]
+    if "call" in report and report["call"].failed:
+        _log("cat /etc/os-release")
+        _log("cat /etc/buildinfo")
+        _log("dmesg -l 5")
+        _log("lsns")
+        _log("ip -br a")
+        _log("ip -br r")
+        _log("ip -br -6 r")
```

Comment:
Could you use long forms here? I tried to look up what the `-b` and `-r` options did separately and could not make sense of it. Instead this seams to be short for `ip -brief address` and `ip -brief route`.
(Using `-6` is fine).

Reply:
Fixed. We are now using the long versions.